main/calculations.py

```diff
@@ -0,0 +1,147 @@
+import requests
+import json
+from datetime import datetime
+from constance import config
+from proxy_requests.proxy_requests import ProxyRequests
+
+from django.core.mail import send_mail
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.contrib import messages
+
+from .models import SiteToCheck
+
+
+def send_email(message, email):
+    subject = 'Errors on my pages'
+    message = message
+    from_email = settings.EMAIL_ADDRESS
+    to_email = email
+    send_mail(
+        subject,
+        message,
+        from_email,
+        [to_email],
+        fail_silently=False,
+    )
+
+
+def my_cron_job():
+    users = User.objects.all()
```

Comment:
Tutaj możesz użyć takiego tricku jak `prefetch_related` To jest już trochę bardziej zaawansowane Django ale w skrócie to Ci pod spodem zrobi joina rabeli Users z tabelą `SiteToCheck` dzięki czemu nie będzie zapytania per pętla. W skrócie zrób tak `users = User.objects.prefetch_related('sitetocheck')`.

Reply:
Myślałem nad tym i raczej prefetch_related nie da się tutaj zastosować. Muszę iterować po userach i po stronach, bo dla każdego usera z osobna tworzę zmienną `output` i wysyłam maila `        if output:
            send_email(output, user.email)`